pallets/parachain-staking/src/lib.rs

```diff
@@ -1532,7 +1532,9 @@ pub mod pallet {
 					for Bond { owner, amount } in state.delegations {
 						let percent = Perbill::from_rational(amount, state.total);
 						let due = percent * amt_due;
```

Comment:
Can't we move the check `!due.is_zero` before computing due & percent ? It would save some useless computation

Reply:
The only other way is to omit the delegation when we compute the rewardable delegators. We previously discussed and decided not to omit entries, since at each round start we are using this field to [compute](https://github.com/PureStake/moonbeam/blob/master/pallets/parachain-staking/src/lib.rs#L1580) the `delegation_count`.

Reply:
ok but if `due.is_zero` it means that `percent` or `amd_due` is zero no, so there is no point in computing `due` ?
Whatever, it was just a simple (maybe useless) optimization

Reply:
I don't think there is anything to optimize here, because the implementation of `Mul<u128> for PerBill` already makes an exception for the case when one of the operands is zero.